dm_cli/dmss_api/model_utils.py

```diff
@@ -1249,11 +1247,12 @@ def deserialize_primitive(data, klass, path_to_item):
                 "string value, please set its type as `type: {}` in your "
                 "spec. That allows the value to be any type. "
             )
+
+            # The string should be in iso8601 datetime format.
+            if not isinstance(data, str) or len(data) < 8:
+                raise ValueError("This is not a datetime")
+            parsed_datetime = datetime.fromisoformat(data)
             if klass == datetime:
```

Comment:
klass is a very strange variable name, but we should deal with that in another PR 😄